hieroplatform/src/main/java/org/hiero/utils/Randomness.java

```diff
@@ -0,0 +1,34 @@
+package org.hiero.utils;
+
+import org.apache.commons.math3.random.MersenneTwister;
+
+public class Randomness {
+    private static Randomness prg;
+    final private MersenneTwister myPrg;
+
+    private Randomness() {
+        this.myPrg = new MersenneTwister();
+    }
+
+    private Randomness(long seed) {
+        this.myPrg = new MersenneTwister(seed);
+    }
+
+    public static Randomness getInstance() {
+        if (prg == null)
+            prg = new Randomness();
+        return prg;
+    }
+
+    public static Randomness getInstance(long seed) {
+        if (prg == null)
```

Comment:
This function is very misleading, since it won't do anything if you have already called getInstance. You may be thinking you are setting the seed, but in fact you are not doing anything.

Let's separate retrieving the instance from intializing the seed; these should be two separate calls.